packages/teams-js/src/internal/communication.ts

```diff
@@ -605,6 +614,65 @@ function updateRelationships(messageSource: Window, messageOrigin: string): void
 
 const handleParentMessageLogger = communicationLogger.extend('handleParentMessage');
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+function retrieveMessageUUIDFromResponse(response: MessageResponse): MessageUUID {
```

Comment:
>retrieveMessageUUIDFromResponse [](http://example.com/codeflow?start=9&length=31)

What are the semantics for this function if the ID that comes in the response isn't present in any of the maps?

Reply:
I'm not sure what you mean by this. What this does is a `byValue` comparison of the uuid coming in from the host to the callback maps we have and returns their `MessageUUID` key for use. This way we have the correct key to perform operations on the map rather than one that is equal in value but not considered the same object.

If there is no uuid being passed in by the `host` due to older versioning on the Host side, we default back to using the `id` as a number to map to the correct UUIDs, since UUIDs are being passed internally.

Reply:
What happens if the incoming message has a uuid but it's not one that we have in any of the maps?

Reply:
The behavior expectation in that case is that there is no callback associated with the MessageRequest, but it still could be a processable Message.

Reply:
So what value will this function return in that case? It's typed as always returning a `MessageUUID` but what does it return if it can't find a matching message in one of the maps?

Reply:
I see your point, I'll update this function so that it can return an undefined. 

Because we don't know if the Host is using an older version of their SDK or a newer version with uuids, this function should always map to a MessageUUID object through the legacyMessageIdsToUuidMap[response.id]` since they are generated together.